lib/index.js

```diff
@@ -1,43 +1,73 @@
+/* eslint global-require: 0 */
+/* eslint no-process-exit: 0 */
+
 'use strict'
 
+const AmqpStats = require('amqp-stats')
 const _ = require('lodash')
 const debug = require('debug')('loopback:component:mq')
-const setupModel = require('./setup-model')
-const Rabbit = require('./rabbit')
-const RabbitTopology = require('./rabbit-topology')()
-
-module.exports = function loopbackComponentMq(app, config) {
-  debug(config.topology)
-  // Store the component configuration
-  _.set(app, 'settings.loopback-component-mq', {
-    options: config.options || {},
-    topology: config.topology || {},
-  })
+const loopback = require('loopback')
+const url = require('url')
+const rabbit = require('rabbot')
+const queueDef = require('./models/rabbitmq.json')
```

Comment:
I think `modelDef` makes more sense here as it's not the definition of a queue but a model.